wrapper/src/main/java/software/amazon/jdbc/HostSpec.java

```diff
@@ -26,7 +27,7 @@
  * An object representing connection info for a given host. Modifiable fields are thread-safe to support sharing this
  * object with the EFM monitor thread.
  */
-public class HostSpec {
+public final class HostSpec {
```

Comment:
Would mind to explain why this class is final? Usually final classes are used in sensitive context like identity, user permissions, etc where altering class behaviour (by inheriting a new class) may create a vulnerability.

Reply:
So if you don't declare this final than a class that extends this can create a constructor that does not initialize all the fields